core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -195,11 +192,19 @@ private void checkApplicationUpdateStatus(Application application) {
                 versionCode = packageInfo.versionCode;
             }
             RudderLogger.logDebug("Current Installed Version: " + versionCode);
+            boolean isLifeCycleEventsAllowed = config.isTrackLifecycleEvents() || config.isRecordScreenViews();
+            if (isLifeCycleEventsAllowed) {
+                application.registerActivityLifecycleCallbacks(this);
+            }
 
             if (previousVersionCode == -1) {
-                // application was not installed previously, Application Installed events
-                RudderLogger.logDebug("Tracking Application Installed");
+                // application was not installed previously, Application Installed event
                 preferenceManager.saveBuildVersionCode(versionCode);
+                // If trackLifeCycleEvents is not allowed then discard the event
+                if (!isLifeCycleEventsAllowed) {
+                    return;
+                }
+                RudderLogger.logDebug("Tracking Application Installed");
```

Comment:
Looks like `Application Installed` and `Application Updated`, will be sent even though `trackLifeCycleEvents` is set  to false, can you verify that once ?

Reply:
It's working fine. When `trackLifeCycleEvents` is set to `false`, I cannot see any lifeCycleEvents (including `Application Installed` and `Application Updated` events) both in the `logcat` and `gw_jobs_1`. When it is set to `true`, lifeCycleEvents is flowing well. Please let me know if this solves the queries.

Reply:
Can you also try sett `recordScreenViews` to true at the same time ?

Reply:
Found the bug: When RecordScreenView is is set to true then also Application Installed event will be called. Fixing that.